crates/lib/src/core/parser/segments/base.rs

```diff
@@ -509,8 +508,8 @@ pub trait Segment: Any + DynEq + DynClone + Debug + CloneSegment + Send + Sync {
     }
 
     /// Group and count fixes by anchor, return dictionary.
-    fn compute_anchor_edit_info(&self, fixes: &Vec<LintFix>) -> AHashMap<Uuid, AnchorEditInfo> {
-        let mut anchor_info = AHashMap::<Uuid, AnchorEditInfo>::new();
+    fn compute_anchor_edit_info(&self, fixes: &Vec<LintFix>) -> AHashMap<u32, AnchorEditInfo> {
+        let mut anchor_info = AHashMap::<u32, AnchorEditInfo>::new();
```

Comment:
We can try replacing it with IntMap now.

Reply:
My worry was overflow problems? Wdyt?

Reply:
yes, because we are using a global counter, there is a chance of overflow. In my view, it would be ideal to have a local context for the file where we could store the counter, arena, and so on. i’m not sure if there are such large SQL codebases, but this issue could be bypassed by using a thread-local variable so that each thread has its own counter. However, this won't solve the problem for the VSCode plugin

Reply:
For thread-local storage, we could use `RefCell<u32>`. For the method with passing context, I think a simple `Cell<u32>` or just `u32` would be sufficient.